controllers/common.go

```diff
@@ -92,3 +102,35 @@ func reconcileDesiredObjects(ctx context.Context, kubeClient client.Client, logg
 	}
 	return nil
 }
+
+func enabledAcceleratedComputeByAgentConfig(ctx context.Context, c client.Client, log logr.Logger) bool {
+	agentResource := getAmazonCloudWatchAgentResource(ctx, c)
+	// missing feature flag means it's on by default
+	featureConfigExists := strings.Contains(agentResource.Spec.Config, acceleratedComputeMetrics)
+	conf, err := adapters.ConfigStructFromJSONString(agentResource.Spec.Config)
+	if err != nil {
+		log.Error(err, "Failed to unmarshall agent configuration")
+		return false
+	}
+
+	if conf.Logs != nil && conf.Logs.LogMetricsCollected != nil && conf.Logs.LogMetricsCollected.Kubernetes != nil {
+		if conf.Logs.LogMetricsCollected.Kubernetes.EnhancedContainerInsights {
+			return !featureConfigExists || conf.Logs.LogMetricsCollected.Kubernetes.AcceleratedComputeMetrics
+		} else {
+			// enhanced container insights is disabled
+			return false
```

Comment:
I think i'm a little confused here...

so the accelerated compute feature is *enabled* if both container insights AND accelerated compute metrics are enabled? 

If that's true can we just return if both those config fields exist?

Reply:
yes, both need to be `true` to create these resources. 

> If that's true can we just return if both those config fields exist?

No really. These 2 cases should be equal:
- missing `accelerated_compute_metric`
- `accelerated_compute_metric` is set to `true`
All other cases return `false`